src/classes/presentation_controller.vala

```diff
@@ -424,30 +547,88 @@ namespace pdfpc {
                     context.fill();
                 }
             }
+            if (drawing_present) {
```

Comment:
Could you draw on a different layer? We could staple all layers correctly then (which means, pointer above the drawing).